### PR TITLE
Don't squash specially named datasources

### PR DESCRIPTION
**Pull request by rbarry82**
If dashboard templates already have `prometheusds` or `lokids`,
leave them alone

### CODE REVIEW

File: tests/unit/test_dashboard_consumer.py [hunk 1]
```diff
@@ -126,6 +126,58 @@
     }
 )
 
+EXISTING_DATASOURCE_DASHBOARD_TEMPLATE = """
```
Comment (sed-i) [inline tests/unit/test_dashboard_consumer.py:129]: These seem to be unused. Missing commit?

Comment (rbarry82) [reply tests/unit/test_dashboard_consumer.py:129]: No, just not added yet -- I wanted you to be able to build to unblock.

Comparing the equality of a list of dicts in Python is absolutely terrible.

**Review by sed-i — APPROVED** [review]
Works. Hopefully merged soon so I won't have to change the load test's cloud-config.

**Comment by mmanciop** [comment]
@rbarry82 I thought we had discussed the logic to be: “If there is precisely one Prometheus datasource, use that whatever its name is”, and to error if a dashboard has TWO datasources of type Prometheus. (And ofc the same logic would work with Loki.) The flexibility in the name of the datasource is very much needed for being able to reuse stock Grafana dashboards.

**Review by sed-i — DISMISSED** [review]

**Comment by rbarry82** [comment]
> @rbarry82 I thought we had discussed the logic to be: “If there is precisely one Prometheus datasource, use that whatever its name is”, and to error if a dashboard has TWO datasources of type Prometheus. (And ofc the same logic would work with Loki.) The flexibility in the name of the datasource is very much needed for being able to reuse stock Grafana dashboards.

The logic in the previous patch was actually "if a dashboard is imported and the name is _not_ `prometheusds|lokids` and it has the appropriate type, rename it so the dropdowns can find it". 

This was added explicitly for the case of "reuse stock Grafana dashboards" where the "template wizard" from the Grafana marketplace isn't used and the JSON is added directly, or in the case of `cos-config`, where it may be "there is a template variable defined for a Prometheus but it doesn't have the name we expect".

No change in behavior other than "don't show annoying error messages in the Grafana UI because your datasource was named `ds_prometheus` or whatever the default from the marketplace is"

**Comment by mmanciop** [comment]
> if a dashboard is imported and the name is not prometheusds|lokids and it has the appropriate type, rename it so the dropdowns can find it

I feel like I am being dense at this stage, but hey I’ll go for broke: do you rename the datasource variable, or do you rename the reference to it in the variable dropdowns with the Juju topology?

**Comment by rbarry82** [comment]
> > if a dashboard is imported and the name is not prometheusds|lokids and it has the appropriate type, rename it so the dropdowns can find it
> 
> I feel like I am being dense at this stage, but hey I’ll go for broke: do you rename the datasource variable, or do you rename the reference to it in the variable dropdowns with the Juju topology?

The reference in the dropdown is always `${prometheusds}` (which was chosen to avoid conflicting with `DS_PROMETHEUS`).

**Review by sed-i — DISMISSED** [review]

**Comment by mmanciop** [comment]
I think this PR should address #84

**Comment by mmanciop** [comment]
I think this PR should address #84

**Comment by rbarry82** [comment]
Closes #84

**Comment by mmanciop** [comment]
> Closes #84

My man, shouldn't the offending dashboard be added to the tests? ;-)

**Review by Abuelodelanada — APPROVED** [review]

**Review by mmanciop — APPROVED** [review]

**Pull request merged**